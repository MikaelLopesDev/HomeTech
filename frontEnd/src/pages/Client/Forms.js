import { Formik } from 'formik';
import React, { useState,useEffect } from 'react';
import { StyleSheet, Text, View,TextInput ,Alert ,ScrollView,TouchableOpacity, SafeAreaView ,Button} from 'react-native';
import NavPrevIcon from '../../assets/nav_prev.svg';
import NavNextIcon from '../../assets/nav_next.svg';
import AsyncStorage from '@react-native-async-storage/async-storage'


import {
    Feather

} from "react-native-vector-icons";



export default function Forms({ navigation }) {
  
  const[service,setService] = useState("");

    const [marca, setMarca] = useState("");
    const [garantia, setGarantia] = useState("");
    const [problema, setProblema] = useState("");
    const [informacoeasAdicionais, setInformacoeasAdicionais] = useState("");
    const [selectedYear,setSelectedYear] = useState(0);
    const [selectedMonth,setSelectedMonth] = useState(0);
    const [selectedDay,setSelectedDay] = useState(0);
    const [selectedHour,setSelectedHour] = useState(null);
    const [listDays, setListDays] = useState([]);
    const [listHours, setListHours] = useState([]);



    async function handleFetchData(){
      const service = await AsyncStorage.getItem("@saveservice:chooosed");
  
      setService(service) 

    }

    useEffect(() => {

      handleFetchData();
    
    },[]);
    



    
    async function handleNew() {
      try {
        await AsyncStorage.setItem("@saveforms:service", service);
        await AsyncStorage.setItem("@saveforms:marca", marca);
        await AsyncStorage.setItem("@saveforms:garantia", garantia);
        await AsyncStorage.setItem("@saveforms:problema", problema);
        await AsyncStorage.setItem("@saveforms:informacoeasAdicionais", informacoeasAdicionais);
        await AsyncStorage.setItem("@saveforms:hour", JSON.stringify(selectedHour));
        await AsyncStorage.setItem("@saveforms:day", JSON.stringify(selectedDay));
        await AsyncStorage.setItem("@saveforms:month", JSON.stringify(selectedMonth));
        await AsyncStorage.setItem("@saveforms:year", JSON.stringify(selectedYear));
    
        // Feedback de sucesso
        Alert.alert(
          "Sucesso",
          "Seu formulário foi enviado com sucesso!",
          [
            {
              text: "OK",
              onPress: () => {
                
                navigation.navigate('TabRoutesClient'); 
              }
            }
          ]
        );
      } catch (error) {
        // Feedback de erro
        Alert.alert(
          "Erro",
          "Houve um problema ao enviar o formulário. Tente novamente.",
          [
            { text: "OK" }
          ]
        );
      }
    }
    

    const months = [ 
      'janeiro',
      'Fevereiro',
      'Março',
      'Abril',
      'Maio',
      'Junho',
      'Julho',
      'Agosto',
      'Setembro',
      'Outubro',
      'Novembro',
      'Dezembro'
      
     ];
    
    
     const days = [
       'Dom',
       'Seg',
       'Ter',
       'Qua',
       'Qui',
       'Sex',
       'Sab'
    
     ];
    
    
    
     const available = [
    
      { id: 1,
        date:"2022-05-10",
        hour:["08:00",
             "09:00",
             "10:00",
             "11:00"]  
    
      },
      {
        id:2,
        date:"2022-05-11",
        hour:["08:00",
             "14:00",
             "15:00",
             "16:00"]  
      },
      {
        id:3,
        date:"2022-05-12",
        hour:["12:00",
             "13:00",
             "14:00",
             "16:00"]  
      },
      {
        id:4,
        date:"2022-05-13",
        hour:["07:00",
             "09:00",
             "14:00",
             "16:00"]  
      },
      {
        id:5,
        date:"2022-05-14",
        hour:["08:00",
             "13:00",
             "15:00",
             "16:00"]  
      },
      {
        id:6,
        date:"2022-05-15",
        hour:["07:00",
             "09:00",
             "13:00",
             "15:00"]   
    
      },
      {
        id:7,
        date:"2022-05-16",
        hour:["13:00",
             "14:00",
             "15:00",
             "16:00"]  
      },
      {
        id:8,
        date:"2022-05-17",
        hour:["07:00",
             "09:00",
             "10:00",
             "12:00"]  
      },
      {
        id:9,
        date:"2022-05-18",
        hour:["08:00",
             "09:00",
             "11:00",
             "12:00"]  
      },
      {
        id:10,
        date:"2022-05-19",
        hour:[
             "09:00",
             "10:00",
             "11:00"]  
      },
      {
        id:11,
        date:"2022-05-20",
        hour:["13:00",
             "15:00",
             "16:00",
             "19:00"]  
      },
      
      
     
    
     ];
    

    useEffect(() => {
      let daysInMonth = new Date(selectedYear, selectedMonth +1, 0).getDate();
      let newListDays = [];


      for(let i=1;i<=daysInMonth;i++) {
        let d = new Date(selectedYear, selectedMonth,i);
        let year = d.getFullYear();
        let month = d.getMonth() + 1;
        let day = d.getDate();
        month = month <10 ? '0'+ month : month;
        day = day < 10? '0'+ day : day;
        let selDate = `${year}-${month}-${day}`;
       
        let avaiability =  available.filter(e=>e.date === selDate);



        newListDays.push({
        status: avaiability.length > 0 && avaiability.length!=null ? true : false,
        weekday: days[d.getDay()],
        number: i
        
        });


      }


      setListDays(newListDays);
      setSelectedDay(1);
      setListHours([]);
      setSelectedHour(0);

    }, [selectedMonth,selectedYear]);


    useEffect(() => {
     if(selectedDay>0){
      let d = new Date(selectedYear, selectedMonth,selectedDay);
      let year = d.getFullYear();
      let month = d.getMonth() + 1;
      let day = d.getDate();
      month = month <10 ? '0'+ month : month;
        day = day < 10? '0'+ day : day;
      let selDate = `${year}-${month}-${day}`;
     
      let avaiability =  available.filter(e=>e.date === selDate);
    
      if(avaiability.length >0){
        setListHours(avaiability[0].hour);
        }
        
     }    

    }, [selectedDay]);

    useEffect(() => {
      let today = new Date();
      setSelectedYear( today.getFullYear());
      setSelectedMonth( today.getMonth());
      setSelectedDay (today.getDate());
   


    }, []);

const handleLeftDateClick = () => {

let mountDate = new Date(selectedYear, selectedMonth,1);
mountDate.setMonth(mountDate.getMonth() - 1);
setSelectedYear(mountDate.getFullYear());
setSelectedMonth(mountDate.getMonth());
setSelectedDay(0);



}
    const handleRightDateClick = () => {
      let mountDate = new Date(selectedYear, selectedMonth,1);
mountDate.setMonth(mountDate.getMonth() + 1);
setSelectedYear(mountDate.getFullYear());
setSelectedMonth(mountDate.getMonth());
setSelectedDay(0);


    }
    
    
      
    
   
    
    
    return(
      
        <Formik>

   <SafeAreaView style={styles.container}>
     <View style={styles.containerTitle}>
   <Text style={styles.Title}>
        Formulário de serviço
      </Text>

     
      </View>
        <View style={styles.inputWrapper} >
             
      


        <View> 
           <Text style={styles.textIncluded} > Serviço selecionado : {service} </Text>
           
           </View>

             <View>
                 
                    <Text style={styles.fieldName}>Qual marca/modelo produto? </Text>
                    <View style={styles.input}>
                      <TextInput
                        style={styles.fieldText}
                        value={marca}
                        onChangeText = {setMarca}
                        
                       
                      />
                    </View>
                  </View>
                  <View>
                    <Text style={styles.fieldName}>Qual problema encontrado ?</Text>
                    <View style={styles.input}>
                      <TextInput
                        style={styles.fieldText}
                        value={problema}
                        onChangeText = {setProblema}
                        
                        
                    
                      />
                    </View>
                  </View>
                  <View>
                    <Text style={styles.fieldName}>O problema está coberto de garantia ?</Text>
                    <View style={styles.input}>
                      <TextInput
                        style={styles.fieldText}
                        value={garantia}
                        onChangeText = {setGarantia}
                                              
                    
                      />
                    </View>
                  </View>
                  <View>
                    <Text style={styles.fieldName}>Informações adicionais:</Text>
                    <View style={styles.input}>
                      <TextInput
                        style={styles.fieldText}
                        value={informacoeasAdicionais}
                        onChangeText = {setInformacoeasAdicionais}
                      
                                            
                      />
                    </View>
                  </View>



                  <View style={styles.dateContainer}>
                
                <View style={styles.dateInfo}>
                

                <View style={styles.dateMontsYears}>
                <TouchableOpacity onPress={ handleLeftDateClick}>
                
                <Feather
                name="chevron-left"
                size={25}
                
              />
                </TouchableOpacity >
                              
                    
                    <Text style={styles.textodateMontsYears} >  {months[selectedMonth]} {selectedYear}  </Text> 
                    
                    
                    <TouchableOpacity onPress={handleRightDateClick}>                  
                     <Feather
                name="chevron-right"
                size={25 }
                
              />
                </TouchableOpacity>

                </View>

              <ScrollView horizontal={true} showHorizontalScrollIndicator={false}>
             
                {listDays.map((item,key)=>(
                   
                   
                   
                   <TouchableOpacity style={{opacity : item.status?1:0.5} && styles.dateItem} key={key}  onPress={()=>setSelectedDay(item.number) }>
              
                <Text style={styles.dateItemWeekDay}>{item.weekday} </Text>

                <Text style={styles.dateItemNumber}> {item.number} </Text>

               </TouchableOpacity>
                   
                ))}
                                      
                 </ScrollView>
              

                </View>

                </View>
                    
                
               
                <View style={styles.TimeInfo}>              
             
             <ScrollView horizontal={true} showHorizontalScrollIndicator={false}>
             
             {listHours.map((item,key)=>(
                
               

                
                <TouchableOpacity style={styles.HourItem} key={key}  onPress={()=> setSelectedHour(item)}>
           

             <Text style={styles.HourNumber}> {item} </Text>
              
            </TouchableOpacity>
                
             ))}
                               
              </ScrollView>
              </View>

  
                  
                  <View style={styles.buttonContainer}>
                  <Button 
                  color="#3B5998"
        title="Enviar"
        onPress={handleNew }
        
        
      />
      </View>
                 
        </View>
        </SafeAreaView>
        </Formik>
        



       

        );
      
       

       

}

const styles = StyleSheet.create({
  containerTitle:{
    height: 100,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#3B5998',
  },
  Title:{

    color: '#FFF',
    fontSize: 21,
    fontWeight: 'bold',
  },

  subTitle:{

fontSize: 15,
color:"#000000",
lineHeight:25,
  },



  container: {
    flex: 1,
    
  },
    input: {
        borderWidth: 1,
        width: "100%",
        height: 24,
        borderRadius: 15,
        justifyContent: "center",
      },
     
      inputWrapper: {
        width: "85%",
        alignSelf: "center",
        
      },
      
      input: {
        borderWidth: 1,
        width: "100%",
        height: 24,
        borderRadius: 6,
        justifyContent: "center",
      },
      textIncluded:{
        marginTop: 30,
        fontFamily: 'sans-serif-medium',
        fontWeight: 'bold',
        fontSize:18,
      },
      fieldName: {
       marginTop: 30,
       fontFamily: 'sans-serif-medium',
       fontWeight: 'bold',
       fontSize:15,

      },

      fieldText: {
     flex: 1,

      },


      dateContainer: {
        
        marginTop: 15,
      },
      
      dateInfo: {
   
  alignItems: "center",
  justifyContent: "center",
  

      },
       
      dateMontsYears: {
  
   
     flexDirection: "row",
  alignItems: "center",
  justifyContent: "center", 
  paddingTop: 15,


      },

      textodateMontsYears: {
        fontSize: 15,
        fontWeight: 'bold',
        
        
      },

      dateItem: {
        width: 45,
        justifyContent: "center",
        borderRadius: 10,
        paddingTop: 5,
        paddingBottom: 5,
        alignItems: "center",
        
        
      },

      dateItemWeekDay: {
        fontSize: 15,
        fontWeight: 'bold',
      },

      dateItemNumber: {
        fontSize: 15,
        fontWeight: 'bold',
     
      },
      TimeInfo: {
   
        alignItems: "center",
        justifyContent: "center",
        
        
      
            },


     

      buttonContainer:{
        marginTop: 20,
       
      },

      HourItem:{
        
      },
      HourNumber:{
        fontSize: 15,
        paddingTop: 6,
   paddingRight: 25,
   fontWeight: 'bold',
      }

          });
