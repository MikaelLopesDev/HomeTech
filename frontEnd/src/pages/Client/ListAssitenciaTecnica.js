import React, { useState, useEffect } from "react";
import { StyleSheet, Text, View, Image,FlatList,TouchableOpacity, SafeAreaView } from 'react-native';
import AsyncStorage from '@react-native-async-storage/async-storage'


export default function ListAssitenciaTecnica({navigation}) {

  const [service, setService] = useState("");


  async function handleNew(){
 
    await AsyncStorage.setItem("@saveservice:chooosed", service);
    
   }

  
 const assistenciaTecnicaList = [

    {
      id:"1",
      
      name:"Aparelho de som", 

    },
    {
        id:"2",
       
        name:"Ar condicionado", 
  
      },

      {
        id:"3",
        name:"Celular", 
  
      },

      {
        id:"4",
        name:"Computador Desktop", 
  
      },

      {
        id:"5",
        name:"Câmera", 
  
      },

      {
        id:"6",
        name:"Fone de ouvido", 
  
      },

      {
        id:"7",
        name:"Eletrodoméstico", 
  
      },

      {
        id:"8",
        name:"Geladeira e freezer", 
  
      },

      {
        id:"9",
        name:"Impressora", 
  
      },

      {
        id:"10",
        name:"Mircro-ondas", 
  
      }, 
      {
        id:"11",
        name:"Relógio", 
  
      },

      {
        id:"12",
        name:"Tablet", 
  
      },
      {
        id:"13",
        name:"Telefone Fixo", 
  
      },

      {
        id:"14",
        name:"Televisão", 
  
      },

      {
        id:"15",
        name:"Vídeo game", 
  
      },

 ]

 let onPressItem = (name) => {

  setService(name);

 }




 const oneService = ( {item} ) => (
 

    <TouchableOpacity  onPress={() =>  onPressItem(item.name)}>
  <View style = {styles.item}>
  <Text style = {styles.name}>{item.name}</Text>
  </View>
  </TouchableOpacity>
  )
  





  headerComponent = () => {

    return <Text style = {styles.lisHeadline}>Serviços Assistência Técnica</Text>
  }
  
  itemSeparator = () => {
    return <View style = {styles.separator} />
  }
  
    return (
      <SafeAreaView >
       
        <FlatList
        ListHeaderComponentStyle = {styles.listHeader}
        ListHeaderComponent={headerComponent}
       data = {assistenciaTecnicaList}
       renderItem = {oneService} 
       
       ItemSeparatorComponent = {itemSeparator}
      
     
       />

<View
                style={{
                  width: "90%",
                  alignSelf: "center",
                  flexDirection: "row",
                  justifyContent: "space-between",
                  marginBottom: 30,
                }}
              >
                <TouchableOpacity
                  style={styles.button}
                  onPress={() => {
                    handleNew()
                    console.log(service)
                    navigation.navigate('Forms')
                  }}
                >
                  <Text
                    style={{ color: "white", fontSize: 20, fontWeight: "700" }}
                  >
                    confirmar
                  </Text>
                </TouchableOpacity>
                
              </View>

       
        </SafeAreaView>
    );
  }
  
  const styles = StyleSheet.create({
    listHeader:{
   height: 70,
   alignItems: 'center',
   justifyContent: 'center',
   backgroundColor: '#3B5998',
   
    },
   lisHeadline:{
  color: '#FFF',
  fontSize: 21,
  fontWeight: 'bold',
  
  
  
   },
   item:{
  flex: 1,
  flexDirection: 'row',
  alignItems: 'center',
  paddingVertical: 10,

  
   },
  
   name:{
  fontWeight: 'bold',
  fontSize:18,
  marginLeft: 15,
  
   },
  
    separator: {
      height: 1,
      backgroundColor: '#FFF',
      width: '100%',
      
    },
     button: {
      height: 38,
      width: 105,
      flex:1,
      backgroundColor: "#3B5998",
      borderRadius: 15,
      alignItems: "center",
      justifyContent: "center",
    },
  });
  