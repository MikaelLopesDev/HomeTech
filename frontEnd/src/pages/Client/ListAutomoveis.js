import React, { useState } from "react";
import { StyleSheet, Text, View, TouchableOpacity, SafeAreaView, FlatList } from 'react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';

export default function ListAssitenciaTecnica({ navigation }) {
  const [selectedId, setSelectedId] = useState(null);

  async function handleNew(selectedService) {
    await AsyncStorage.setItem("@saveservice:chooosed", selectedService);
  }

 const assistenciaTecnicaList = [

    {
        
            id:1,
            name:"Alarme automotivo", 
      
          },
          {
              id:2,
              name:"Ar condicionado", 
        
            },
      
            {
              id:3,
              name:"Auto elétrico", 
        
            },
      
            {
              id:4,
              name:"Borracharia", 
        
            },
      
            {
              id:5,
              name:"Guincho", 
        
            },
      
            {
              id:6,
              name:"Higienização e Polimento", 
        
            },
      
            {
              id:7,
              name:"Insulfilm", 
        
            },
      
            {
              id:8,
              name:"Martelinho de ouro", 
        
            },
      
            {
              id:9,
              name:"Pintura", 
        
            },
      
            {
              id:10,
              name:"Som automotivo", 
        
            }, 
            {
              id:11,
              name:"Vendas de Automóveis", 
        
            },
      
            {
              id:12,
              name:"Vidraçaria", 
        
            },

 ]

 
 const onPressItem = (id) => {
  setSelectedId(id);
};

const oneService = ({ item }) => (
  <TouchableOpacity onPress={() => onPressItem(item.id)}>
    <View style={[styles.item, selectedId === item.id && styles.selectedItem]}>
      <Text style={styles.name}>{item.name}</Text>
    </View>
  </TouchableOpacity>
);

const headerComponent = () => {
  return <Text style={styles.lisHeadline}>Serviços Assistência Técnica</Text>;
};

const itemSeparator = () => {
  return <View style={styles.separator} />;
};

return (
  <SafeAreaView>
    <FlatList
      ListHeaderComponentStyle={styles.listHeader}
      ListHeaderComponent={headerComponent}
      data={assistenciaTecnicaList}
      renderItem={oneService}
      ItemSeparatorComponent={itemSeparator}
      keyExtractor={item => item.id.toString()}
    />

    <View style={styles.buttonContainer}>
      <TouchableOpacity
        style={styles.button}
        onPress={() => {
          const selectedService = assistenciaTecnicaList.find(item => item.id === selectedId)?.name;
          if (selectedService) {
            handleNew(selectedService);
            console.log(selectedService);
            navigation.navigate('Forms');
          }
        }}
      >
        <Text style={styles.buttonText}>Confirmar</Text>
      </TouchableOpacity>
    </View>
  </SafeAreaView>
);
}

const styles = StyleSheet.create({
listHeader: {
  height: 70,
  alignItems: 'center',
  justifyContent: 'center',
  backgroundColor: '#3B5998',
},
lisHeadline: {
  color: '#FFF',
  fontSize: 21,
  fontWeight: 'bold',
},
item: {
  flex: 1,
  flexDirection: 'row',
  alignItems: 'center',
  paddingVertical: 10,
},
selectedItem: {
  backgroundColor: '#3B5998', // Cor quando o item é selecionado
},
name: {
  fontWeight: 'bold',
  fontSize: 18,
  marginLeft: 15,
},
separator: {
  height: 1,
  backgroundColor: '#FFF',
  width: '100%',
},
buttonContainer: {
  width: "90%",
  alignSelf: "center",
  flexDirection: "row",
  justifyContent: "space-between",
  marginBottom: 30,
},
button: {
  height: 38,
  width: 105,
  flex: 1,
  backgroundColor: "#3B5998",
  borderRadius: 15,
  alignItems: "center",
  justifyContent: "center",
},
buttonText: {
  color: "white",
  fontSize: 20,
  fontWeight: "700",
},
});