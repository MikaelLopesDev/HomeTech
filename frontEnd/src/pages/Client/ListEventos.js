import React, { useState, useEffect } from "react";
import { StyleSheet, Text, View, Image,FlatList,TouchableOpacity, SafeAreaView } from 'react-native';
import AsyncStorage from '@react-native-async-storage/async-storage'




export default function ListAssitenciaTecnica({navigation}) {

  const [service, setService] = useState("");

  async function handleNew(){
 
    await AsyncStorage.setItem("@saveservice:chooosed", service);
    
   }
  
 const assistenciaTecnicaList = [

    {
        id:1,
        name:"Aparelho Animação de festas", 
  
      },
      {
          id:2,
          name:"Acessor de Eventos", 
    
        },
  
        {
          id:3,
          name:"Bandas e cantores", 
    
        },
  
        {
          id:4,
          name:"Bartenders", 
    
        },
  
        {
          id:5,
          name:"Brindes e lembrancinhas", 
    
        },
  
        {
          id:6,
          name:"Buffet completo", 
    
        },
  
        {
          id:7,
          name:"Churrasqueiro", 
    
        },
  
        {
          id:8,
          name:"Confeteira", 
    
        },
  
        {
          id:9,
          name:"Decoração", 
    
        },
  
        {
          id:10,
          name:"Djs", 
    
        }, 
        {
          id:11,
          name:"Confeitaria", 
    
        },
  
        {
          id:12,
          name:"Florista", 
    
        },
        {
          id:13,
          name:"Garçons", 
    
        },

 ]

 let onPressItem = (name) => {

  setService(name)
 
  
  
 }

 const oneService = ( {item} ) => (
  
    <TouchableOpacity  onPress={() => onPressItem(item.name)}>
  <View style = {styles.item}>
  <Text style = {styles.name}>{item.name}</Text>
  </View>
  </TouchableOpacity>
  )
  





  headerComponent = () => {

    return <Text style = {styles.lisHeadline}>Serviços Assistência Técnica</Text>
  }
  
  itemSeparator = () => {
    return <View style = {styles.separator} />
  }
  
    return (
      <SafeAreaView >
       
        <FlatList
        ListHeaderComponentStyle = {styles.listHeader}
        ListHeaderComponent={headerComponent}
       data = {assistenciaTecnicaList}
       renderItem = {oneService} 
       
       ItemSeparatorComponent = {itemSeparator}
      
     
       />


<View
                style={{
                  width: "90%",
                  alignSelf: "center",
                  flexDirection: "row",
                  justifyContent: "space-between",
                  marginBottom: 30,
                }}
              >
                <TouchableOpacity
                  style={styles.button}
                  onPress={() => {
                    handleNew()
                    console.log(service)
                    navigation.navigate('Forms')
                  }}
                >
                  <Text
                    style={{ color: "white", fontSize: 20, fontWeight: "700" }}
                  >
                    confirmar
                  </Text>
                </TouchableOpacity>
                
              </View>

        </SafeAreaView>
    );
  }
  
  const styles = StyleSheet.create({
    listHeader:{
   height: 70,
   alignItems: 'center',
   justifyContent: 'center',
   backgroundColor: '#3B5998',
   
    },
   lisHeadline:{
  color: '#FFF',
  fontSize: 21,
  fontWeight: 'bold',
  
  
  
   },
   item:{
  flex: 1,
  flexDirection: 'row',
  alignItems: 'center',
  paddingVertical: 10,

  
   },
  
   name:{
  fontWeight: 'bold',
  fontSize:18,
  marginLeft: 15,
  
   },
  
    separator: {
      height: 1,
      backgroundColor: '#FFF',
      width: '100%',
      
    },
    button: {
      height: 38,
      width: 105,
      flex:1,
      backgroundColor: "#3B5998",
      borderRadius: 15,
      alignItems: "center",
      justifyContent: "center",
    },

  });
  