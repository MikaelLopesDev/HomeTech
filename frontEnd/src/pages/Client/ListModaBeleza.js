import React, { useState, useEffect } from "react";
import { StyleSheet, Text, View, Image,FlatList,TouchableOpacity, SafeAreaView } from 'react-native';
import AsyncStorage from '@react-native-async-storage/async-storage'


export default function ListAssitenciaTecnica({navigation}) {

  const [service, setService] = useState("");


  async function handleNew(){
 
    await AsyncStorage.setItem("@saveservice:chooosed", service);
    
   }

 const assistenciaTecnicaList = [

    {
        id:1,
        name:"Afaiate", 
  
      },
      {
          id:2,
          name:"Artesanato", 
    
        },
  
        {
          id:3,
          name:"Barbeiro", 
    
        },
  
        {
          id:4,
          name:" Bronzeamento", 
    
        },
  
        {
          id:5,
          name:"Cabelereiro", 
    
        },
  
        {
          id:6,
          name:" cortes e costura", 
    
        },
  
        {
          id:7,
          name:"Depilação", 
    
        },
  
        {
          id:8,
          name:" Designer de Cílios", 
    
        },
  
        {
          id:9,
          name:"Esteticista", 
    
        },
  
        {
          id:10,
          name:"Manicure e pedicure", 
    
        }, 
        {
          id:11,
          name:"Sapateiro", 
    
        },
  
        {
          id:12,
          name:"Maquiadores", 
    
        },
        {
          id:13,
          name:"Micropigmentador", 
    
        },

 ]

 let onPressItem = (name) => {

  setService(name)
 
  
 }

 const oneService = ( {item} ) => (
  
    <TouchableOpacity  onPress={() => onPressItem(item.name)}>
  <View style = {styles.item}>
  <Text style = {styles.name}>{item.name}</Text>
  </View>
  </TouchableOpacity>
  )
  





  headerComponent = () => {

    return <Text style = {styles.lisHeadline}>Serviços Assistência Técnica</Text>
  }
  
  itemSeparator = () => {
    return <View style = {styles.separator} />
  }
  
    return (
      <SafeAreaView >
       
        <FlatList
        ListHeaderComponentStyle = {styles.listHeader}
        ListHeaderComponent={headerComponent}
       data = {assistenciaTecnicaList}
       renderItem = {oneService} 
       
       ItemSeparatorComponent = {itemSeparator}
      

       
     
       />


<View
                style={{
                  width: "90%",
                  alignSelf: "center",
                  flexDirection: "row",
                  justifyContent: "space-between",
                  marginBottom: 30,
                }}
              >
                <TouchableOpacity
                  style={styles.button}
                  onPress={() => {
                    handleNew()
                    console.log(service)
                    navigation.navigate('Forms')
                  }}
                >
                  <Text
                    style={{ color: "white", fontSize: 20, fontWeight: "700" }}
                  >
                    confirmar
                  </Text>
                </TouchableOpacity>
                
              </View>
        </SafeAreaView>
    );
  }
  
  const styles = StyleSheet.create({
    listHeader:{
   height: 70,
   alignItems: 'center',
   justifyContent: 'center',
   backgroundColor: '#3B5998',
   
    },
   lisHeadline:{
  color: '#FFF',
  fontSize: 21,
  fontWeight: 'bold',
  
  
  
   },
   item:{
  flex: 1,
  flexDirection: 'row',
  alignItems: 'center',
  paddingVertical: 10,

  
   },
  
   name:{
  fontWeight: 'bold',
  fontSize:18,
  marginLeft: 15,
  
   },
  
    separator: {
      height: 1,
      backgroundColor: '#FFF',
      width: '100%',
      
    },
    button: {
      height: 38,
      width: 105,
      flex:1,
      backgroundColor: "#3B5998",
      borderRadius: 15,
      alignItems: "center",
      justifyContent: "center",
    },
  });
  