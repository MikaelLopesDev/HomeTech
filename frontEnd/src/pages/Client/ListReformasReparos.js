import React, { useState } from "react";
import { StyleSheet, Text, View, TouchableOpacity, SafeAreaView, FlatList } from 'react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';

export default function ListAssitenciaTecnica({ navigation }) {
  const [selectedId, setSelectedId] = useState(null);

  async function handleNew(selectedService) {
    await AsyncStorage.setItem("@saveservice:chooosed", selectedService);
  }

 const assistenciaTecnicaList = [

    {
        id:1,
        name:"Acessibilidade", 
  
      },
      {
          id:2,
          name:"Afiaçção", 
    
        },
  
        {
          id:3,
          name:"Agrimensura", 
    
        },
  
        {
          id:4,
          name:"Arquiteto", 
    
        },
  
        {
          id:5,
          name:"Banheira", 
    
        },
  
        {
          id:6,
          name:"Casas e Chalés de madeira", 
    
        },
  
        {
          id:7,
          name:"Chaveiro", 
    
        },
  
        {
          id:8,
          name:" Cimatização ", 
    
        },
  
        {
          id:9,
          name:"Decorador ", 
    
        },
  
        {
          id:10,
          name:"Demolição", 
    
        }, 
        {
          id:11,
          name:"Desentupidor", 
    
        },
  
        {
          id:12,
          name:"Design de interiores", 
    
        },
        {
          id:13,
          name:" Eletricista", 
    
        },
  
        {
          id:14,
          name:"Engenheiro", 
    
        },
  
        {
          id:15,
          name:"Encanador ", 
    
        },
  
        {
        id: 16,
          name : "Fossa ", 
    
        },
        {
          id: 17,
            name : "Gesso e drywall ", 
      
          },
          {
              id: 18,
                name : "Gás", 
          
              },
              {
                  id: 19,
                    name : "Jardinagem ", 
              
                  },
  
                  {
                      id: 20,
                        name : "Marceneiro ", 
                  
                      },
                      {
                          id: 21,
                            name : "Pintor ", 
                      
                          },
                      
                          {
                              id: 22,
                                name : "Piscina ", 
                          
                              },
                      
                              {
                                  id: 23,
                                    name : "Portão automático ", 
                              
                                  },
                      
                                  {
                                      id: 24,
                                        name : "Segurança eletrônica ", 
                                  
                                      },
                                      

 ]
 const onPressItem = (id) => {
  setSelectedId(id);
};

const oneService = ({ item }) => (
  <TouchableOpacity onPress={() => onPressItem(item.id)}>
    <View style={[styles.item, selectedId === item.id && styles.selectedItem]}>
      <Text style={styles.name}>{item.name}</Text>
    </View>
  </TouchableOpacity>
);

const headerComponent = () => {
  return <Text style={styles.lisHeadline}>Serviços Assistência Técnica</Text>;
};

const itemSeparator = () => {
  return <View style={styles.separator} />;
};

return (
  <SafeAreaView>
    <FlatList
      ListHeaderComponentStyle={styles.listHeader}
      ListHeaderComponent={headerComponent}
      data={assistenciaTecnicaList}
      renderItem={oneService}
      ItemSeparatorComponent={itemSeparator}
      keyExtractor={item => item.id.toString()}
    />

    <View style={styles.buttonContainer}>
      <TouchableOpacity
        style={styles.button}
        onPress={() => {
          const selectedService = assistenciaTecnicaList.find(item => item.id === selectedId)?.name;
          if (selectedService) {
            handleNew(selectedService);
            console.log(selectedService);
            navigation.navigate('Forms');
          }
        }}
      >
        <Text style={styles.buttonText}>Confirmar</Text>
      </TouchableOpacity>
    </View>
  </SafeAreaView>
);
}

const styles = StyleSheet.create({
listHeader: {
  height: 70,
  alignItems: 'center',
  justifyContent: 'center',
  backgroundColor: '#3B5998',
},
lisHeadline: {
  color: '#FFF',
  fontSize: 21,
  fontWeight: 'bold',
},
item: {
  flex: 1,
  flexDirection: 'row',
  alignItems: 'center',
  paddingVertical: 10,
},
selectedItem: {
  backgroundColor: '#3B5998', // Cor quando o item é selecionado
},
name: {
  fontWeight: 'bold',
  fontSize: 18,
  marginLeft: 15,
},
separator: {
  height: 1,
  backgroundColor: '#FFF',
  width: '100%',
},
buttonContainer: {
  width: "90%",
  alignSelf: "center",
  flexDirection: "row",
  justifyContent: "space-between",
  marginBottom: 30,
},
button: {
  height: 38,
  width: 105,
  flex: 1,
  backgroundColor: "#3B5998",
  borderRadius: 15,
  alignItems: "center",
  justifyContent: "center",
},
buttonText: {
  color: "white",
  fontSize: 20,
  fontWeight: "700",
},
});