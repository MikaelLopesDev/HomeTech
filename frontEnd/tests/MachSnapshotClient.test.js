import 'react-native';
import React from 'react';
import ListAssitenciaTecnica from '../src/pages/Client/ListAssitenciaTecnica';
import ListReformasReparos from '../src/pages/Client/ListReformasReparos';
import ListModaBeleza from '../src/pages/Client/ListModaBeleza'
import ListEventos from '../src/pages/Client/ListEventos'
import ListAutomoveis from '../src/pages/Client/ListAutomoveis'
import Forms from '../src/pages/Client/Forms';


import renderer from 'react-test-renderer';




it('renders correctly Forms ', () => {

    const tree =  renderer.create(<Forms />).toJSON;
    expect(tree).toMatchSnapshot();

})

it('renders correctly ListAssistenciaTecnica ', () => {

    const tree =  renderer.create(<ListAssitenciaTecnica />).toJSON;
    expect(tree).toMatchSnapshot();

})

it('renders correctly ListAutomoveis ', () => {

    const tree =  renderer.create(<ListAutomoveis />).toJSON;
    expect(tree).toMatchSnapshot();

})

it('renders correctly ListReformasReparos ', () => {

    const tree =  renderer.create(<ListReformasReparos />).toJSON;
    expect(tree).toMatchSnapshot();

})

it('renders correctly ListEventos ', () => {

    const tree =  renderer.create(<ListEventos />).toJSON;
    expect(tree).toMatchSnapshot();

})

it('renders correctly ListModaBeleza ', () => {

    const tree =  renderer.create(<ListModaBeleza />).toJSON;
    expect(tree).toMatchSnapshot();

})





