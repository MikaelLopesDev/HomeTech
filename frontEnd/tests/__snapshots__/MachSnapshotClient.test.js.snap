// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Configurações  1`] = `[Function]`;

exports[`renders correctly Forms  1`] = `[Function]`;

exports[`renders correctly ListAssistenciaTecnica  1`] = `[Function]`;

exports[`renders correctly ListAutomoveis  1`] = `[Function]`;

exports[`renders correctly ListEventos  1`] = `[Function]`;

exports[`renders correctly ListModaBeleza  1`] = `[Function]`;

exports[`renders correctly ListReformasReparos  1`] = `[Function]`;
